tests/test_utils.py

```diff
@@ -1684,6 +1684,33 @@ def test_get_env_default(self):
             "PAASTA_RESOURCE_MEM": "4096",
         }
 
+    def test_get_env_image_version(self):
+        fake_conf = utils.InstanceConfig(
+            service="fake_service",
+            cluster="fake_cluster",
+            instance="fake_instance",
+            config_dict={},
+            branch_dict={
+                "desired_state": "start",
+                "force_bounce": "12345",
+                "docker_image": "something",
+                "git_sha": "9",
+                "image_version": "extrastuff",
+            },
+        )
+        assert fake_conf.get_env() == {
+            "PAASTA_SERVICE": "fake_service",
+            "PAASTA_INSTANCE": "fake_instance",
+            "PAASTA_CLUSTER": "fake_cluster",
+            "PAASTA_DEPLOY_GROUP": "fake_cluster.fake_instance",
+            "PAASTA_GIT_SHA": "somethin",
```

Comment:
Since we currently get git_sha from docker_url, this test is failing.

You'll have to mock `paasta_tools.utils.get_service_docker_registry`, just like test_get_env_with_config.

As an aside, this means that get_git_sha_from_docker_url will also be returning invalid information for images with image_version/extrastuff -- we are just splitting by '-' and returning the last part right now: https://github.com/Yelp/paasta/blob/master/paasta_tools/utils.py#L3456-L3463